### PR TITLE
[tasks] Enabling copyrights linting in CI pipeline

### PR DESCRIPTION
**Pull request by carlosroman**
### What does this PR do?

This PR enables the linter that checks all files have the correct header/copyright. The PR also adds the copyright to the files that had it missing as well.

### Motivation

Automation of copyright header linting.

### Additional Notes

I've excluded some auto generated files from the linter.

### Possible Drawbacks / Trade-offs

Due to the merge freeze this might cause some builds to break once merged.

### Describe how to test/QA your changes

```sh-session
$ inv lint-copyrights
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

**Review by brycekahle — APPROVED** [review]

File: tasks/libs/copyright.py [hunk 1]
```diff
@@ -29,9 +29,15 @@
     '/pkg/security/probe/custom_events_easyjson.go',
     '/pkg/security/probe/serializers_easyjson.go',
     '/pkg/security/secl/model/accessors.go',
+    '/pkg/proto/msgpgo/.*_gen(_test){,1}.go',
+    '/pkg/trace/pb/.*_gen(_test){,1}.go',
+    '/pkg/security/probe/fields_resolver.go',
     # These are files that we should not add our copyright to
     '/third_party/golang/',
     '/third_party/kubernetes/',
+    '/patch/grpc-go-insecure/',
```
Comment (mx-psi) [inline tasks/libs/copyright.py:38]: Did you mean this instead?
```suggestion
    '/internal/patch/grpc-go-insecure/',
```

File: pkg/serverless/invocationlifecycle/rand.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Unless explicitly stated otherwise all files in this repository are licensed
 // under the Apache License Version 2.0.
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
-// Copyright 2016 Datadog, Inc.
+// Copyright 2022-present Datadog, Inc.
```
Comment (hush-hush) [inline pkg/serverless/invocationlifecycle/rand.go:4]: should it be ?
```suggestion
// Copyright 2016-present Datadog, Inc.
```

**Review by hush-hush — APPROVED** [review]
Added a quick comment, looks good overall.

**Review by ahmed-mez — APPROVED** [review]

File: pkg/serverless/invocationlifecycle/rand.go (hunk 2 above)
Comment (carlosroman) [reply pkg/serverless/invocationlifecycle/rand.go:4]: Yes, that was an accident on my part. Will fix it 👍🏽

File: tasks/libs/copyright.py (hunk 1 above)
Comment (carlosroman) [reply tasks/libs/copyright.py:38]: Both work but as it a simple regex. For consistency I'll update it to have the full package path.

Comment (mx-psi) [reply tasks/libs/copyright.py]: Right, didn't realize it was a regex :) both make sense then!

**Review by mx-psi — APPROVED** [review]

**Review by xornivore — APPROVED** [review]
Thank you!

File: tasks/libs/copyright.py [hunk 3]
```diff
@@ -22,17 +22,26 @@
...
     # These are files that we should not add our copyright to
-    '/third_party/golang/',
-    '/third_party/kubernetes/',
+    '/internal/patch/grpc-go-insecure/',
+    '/internal/patch/logr/funcr/funcr(_test){,1}.go',
+    '/internal/patch/logr/funcr/internal/logr/',
+    '/internal/third_party/golang/',
+    '/internal/third_party/kubernetes/',
 ]
+# These path patterns are excluded from checks
```
Comment (vickenty) [inline tasks/libs/copyright.py:44]: I think this comment made more sense before the list 😄

**Review by vickenty — APPROVED** [review]

File: tasks/libs/copyright.py (hunk 3 above)
Comment (carlosroman) [reply tasks/libs/copyright.py:44]: Doh!  Not sure how that ended up there 😅

**Pull request merged**